homeassistant/components/system_bridge/sensor.py

```diff
@@ -80,861 +271,251 @@ async def async_setup_entry(
-    """Defines a Processes CPU System Load sensor."""
-
-    def __init__(
-        self, coordinator: SystemBridgeDataUpdateCoordinator, index: int
-    ) -> None:
-        """Initialize System Bridge sensor."""
-        super().__init__(
-            coordinator,
-            f"processes_load_cpu_{index}_system",
-            f"System Load CPU {index}",
-            "mdi:percent",
-            None,
-            PERCENTAGE,
-            False,
-        )
-        self._index = index
-
-    @property
-    def state(self) -> float | None:
-        """Return the state of the sensor."""
-        bridge: Bridge = self.coordinator.data
-        return (
-            round(bridge.processes.load.cpus[self._index].loadSystem, 2)
-            if bridge.processes.load.cpus[self._index].loadSystem is not None
-            else None
-        )
-
-
-class SystemBridgeProcessesCpuIdleLoadSensor(SystemBridgeSensor):
-    """Defines a Processes CPU Idle Load sensor."""
-
-    def __init__(
-        self, coordinator: SystemBridgeDataUpdateCoordinator, index: int
-    ) -> None:
-        """Initialize System Bridge sensor."""
-        super().__init__(
-            coordinator,
-            f"processes_load_cpu_{index}_idle",
-            f"Idle Load CPU {index}",
-            "mdi:percent",
-            None,
-            PERCENTAGE,
-            False,
-        )
-        self._index = index
-
-    @property
-    def state(self) -> float | None:
-        """Return the state of the sensor."""
-        bridge: Bridge = self.coordinator.data
-        return (
-            round(bridge.processes.load.cpus[self._index].loadIdle, 2)
-            if bridge.processes.load.cpus[self._index].loadIdle is not None
-            else None
-        )
-
-
-class SystemBridgeBiosVersionSensor(SystemBridgeSensor):
-    """Defines a bios version sensor."""
-
-    def __init__(self, coordinator: SystemBridgeDataUpdateCoordinator) -> None:
-        """Initialize System Bridge sensor."""
-        super().__init__(
-            coordinator,
-            "bios_version",
-            "BIOS Version",
-            "mdi:chip",
-            None,
-            None,
-            False,
-        )
-
-    @property
-    def native_value(self) -> str:
-        """Return the state of the sensor."""
-        bridge: Bridge = self.coordinator.data
-        return bridge.system.bios.version
-
-
-class SystemBridgeVersionSensor(SystemBridgeSensor):
-    """Defines a version sensor."""
-
-    def __init__(self, coordinator: SystemBridgeDataUpdateCoordinator) -> None:
-        """Initialize System Bridge sensor."""
+        """Initialize."""
         super().__init__(
             coordinator,
-            "version",
-            "Version",
-            "mdi:counter",
-            None,
-            None,
-            True,
+            description.key,
+            description.name,
+            description.icon,
+            description.enabled_by_default,
```

Comment:
Usually it's enough to set the entity description attribute on the entity. The Entity base class will look up the description for name, icon and enabled by default, etc. Only if we need some custom logic applied onto the description, we need to pass the parameter explicitly, eg for name in this case. Eg:

https://github.com/home-assistant/core/blob/6eadc0c3033473381cb70babf9f7c813e183d2df/homeassistant/helpers/entity.py#L399-L406